http-clients/src/main/java/feign/OptionalAwareDecoder.java

```diff
@@ -35,11 +36,19 @@ public OptionalAwareDecoder(Decoder delegate) {
 
     @Override
     public Object decode(Response response, Type type) throws IOException, FeignException {
-        if (response.status() == 204 && Types.getRawType(type).equals(Optional.class)) {
-            return Optional.absent();
+        if (Types.getRawType(type).equals(Optional.class)) {
+            if (response.status() == 204) {
+                return Optional.absent();
+            } else {
+                return Optional.of(delegate.decode(response, getInnerType(type)));
```

Comment:
I would say if `delegate.decode` returns null, the response code should have been a 204.

Should I change line 43 to `ofNullable`?